addons/mail/static/src/js/models/threads/abstract_thread.js

```diff
@@ -29,6 +35,15 @@ var AbstractThread = Class.extend({
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * Add a message to this thread.
+     *
+     * @param {mail.model.AbstractMessage} message
+     */
+    addMessage: function (message) {
+        this._addMessage.apply(this, arguments);
+        this.trigger('message_added', message);
```

Comment:
This is a trigger so that it acts like a hook. This is useful to add the "typing" feature on threads like a plugin.

Not sure it is worth using 'trigger_up', because it may wrongly assume that someone in the hierarchy hooks on this OdooEvent, which is not the case.